apps/backend/src/modules/users/user.service.ts

```diff
@@ -77,11 +86,31 @@ class UserService implements Service {
 		return await this.userRepository.getByEmail(email);
 	}
 
+	public async getByNickname(nickname: string): Promise<UserEntity | null> {
+		return await this.userRepository.getByNickname(nickname);
+	}
+
 	public async update(
 		userId: number,
 		userProfile: UserProfileRequestDto,
 	): Promise<UserAuthResponseDto | null> {
-		const updatedUser = await this.userRepository.update(userId, userProfile);
+		const { firstName, lastName, nickname } = userProfile;
+		const user = await this.userRepository.getByNickname(nickname);
+		const hasUser = Boolean(user);
+		const isSameUser = Boolean(user?.toObject().id === userId);
```

Comment:
```suggestion
		const isSameUser =user?.toObject().id === userId;
```
It's already Boolean